map_conversion.py

```diff
@@ -0,0 +1,250 @@
+            continue
+        node = ET.SubElement(root, "way")
+
+        if way["token"] not in way_to_lane_id:
+            way_to_lane_id[way["token"]] = global_way_id
+            global_way_id += 1
+        curr_id = way_to_lane_id[way["token"]]
+
+        node.set("lane_id", str(curr_id))
+        traffic = ET.SubElement(node, "tag")
+        traffic.set("k", "has_traffic_control")
+        traffic.set("v", "False")
+
+        turn = ET.SubElement(node, "tag")
+        turn.set("k", "turn_direction")
+        turn.set("v", "NONE")
+
+        intersection = ET.SubElement(node, "tag")
+        intersection.set("k", "is_intersection")
+        intersection.set("v", "False")
+
+        ln = ET.SubElement(node, "tag")
+        ln.set("k", "l_neighbor_id")
+        ln.set("v", "None")
+
+        rn = ET.SubElement(node, "tag")
+        rn.set("k", "r_neighbor_id")
+        rn.set("v", "None")
+
+        for waypoint in lane_dict[way["token"]]:
+            nd = ET.SubElement(node, "nd")
+            nd.set("ref", str(waypoint))
+
+        predecessors = nusc_map.get_incoming_lane_ids(way["token"])
+        successors = nusc_map.get_outgoing_lane_ids(way["token"])
+
+        for pred_id in predecessors:
+            pre = ET.SubElement(node, "tag")
+            pre.set("k", "predecessor")
+            if pred_id not in way_to_lane_id:
+                way_to_lane_id[pred_id] = global_way_id
+                global_way_id += 1
+            int_pred_id = way_to_lane_id[pred_id]
+            pre.set("v", str(int_pred_id))
+
+        for succ_id in successors:
+            succ = ET.SubElement(node, "tag")
+            succ.set("k", "successor")
+            if succ_id not in way_to_lane_id:
+                way_to_lane_id[succ_id] = global_way_id
+                global_way_id += 1
+            int_succ_id = way_to_lane_id[succ_id]
+            succ.set("v", str(int_succ_id))
+
+        lane_id = way_to_lane_id[way["token"]]
+        tableidx_to_laneid_map[table_idx_counter] = lane_id
+        table_idx_counter += 1
+
+        xmin, ymin, xmax, ymax = compute_point_cloud_bbox(
+            poly_dict[way["polygon_token"]]
+        )
+        halluc_bbox_table += [(xmin, ymin, xmax, ymax)]
+
+    halluc_bbox_table = np.array(halluc_bbox_table)
+    halluc_bbox_dict = {
+        "tableidx_to_laneid_map": tableidx_to_laneid_map,
+        "halluc_bbox_table": halluc_bbox_table,
+    }
+    np.save(
+        f"{argo_dir}/{filename_to_id[filename]}_halluc_bbox_table.npy",
+        halluc_bbox_table,
+    )
+    with open(
+        f"{argo_dir}/{filename_to_id[filename]}_tableidx_to_laneid_map.json", "w"
+    ) as outfile:
+        json.dump(tableidx_to_laneid_map, outfile)
+
+    tree = ET.ElementTree(root)
+    with open(
+        f"{argo_dir}/pruned_nuscenes_{filename_to_id[filename]}_vector_map.xml", "wb"
+    ) as files:
+        tree.write(files)
+
+
+def convert_map(args: str) -> None:
```

Comment:
I think `args` is probably of type `CfgNode` not `str`, but you can double check with mypy

Reply:
I saw another script where it was argparse.Namespace, so I have changed to that. Would you like to to check with mypy aswell ?

Reply:
should be right now, thanks